### ISSUE TITLE
Is this use able with python3?

### ISSUE THREAD

**Issue by rjb25** [ericsomdahl/python-bittrex#19]
If so how would I install it?
`setup.py` says python version 2.7 and `pip3` has no package python-bittrex.

**Comment by rjb25** [comment]
`pip3` install bittrex worked

**Comment by ericsomdahl** [comment]
We only support 2.7 but that is only due to my laziness.  PRs are welcome if anyone wants formal support :)

**Comment by avbanks** [comment]
The last commit for the bittrex.py file says "some improvements and python3 compat". I'm assuming it is compatible.

**Comment by nateGeorge** [comment]
Yeah, when I tried installing manually for anaconda3 on Ubuntu linux, it gave the error:

```
from bittrex import Bittrex
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-0033a31ada1b> in <module>()
----> 1 from bittrex import Bittrex

ImportError: cannot import name 'Bittrex'
```

but then I used the pip from anaconda to install, and it works.

**Comment by nateGeorge** [comment]
Seems like the `from .bittrex import *` is missing from __init__.py when I install it using setup.py.

**Comment by biophil** [comment]
@nateGeorge, have you actually tried using it on any python3? I'm getting the classic `module 'urllib' has no attribute 'urlencode'` error on my 3.5 when I try to do any API calls.

**Comment by nateGeorge** [comment]
You have to clone the repo and install it with setup.py; you can't use pip.  The version on pip is too old right now.

**Comment by durika** [comment]
Hi, this is not really compatible with Python3, but there you can fix it by locating the bittrex.py and modifying the line 62 and 64 to following
`		request_url += urllib.parse.urlencode(options)
		
		signature = hmac.new(self.api_secret.encode('utf-8'), request_url.encode('utf-8'), hashlib.sha512).hexdigest()
`

Code to initialize and test:
`from bittrex import Bittrex
api = Bittrex('xxx', 'xxx')
api.get_markets()`

**Comment by harwee** [comment]
I have forked the repo as it was missing get orders and  been using this with python 3. This, is compatible with python3.

You need to import it by 
`from bittrex.bitttex import Bittrex`

**Comment by skyl** [comment]
I was about to make an issue about the `from bittrex.bitttex import Bittrex` and maybe put in a PR to import the class into the __init__.py file. But, that small change gives you:

```
In [1]: import bittrex

In [2]: bittrex.Bittrex
Out[2]: bittrex.bittrex.Bittrex

In [3]: from bittrex import Bittrex

In [4]: Bittrex
Out[4]: bittrex.bittrex.Bittrex
```

Tolerable. I guess you have to modify the setup.py if you want to shake out that extra bittrex.

> You have to clone the repo and install it with setup.py; you can't use pip.

You can install from git with pip though, FYI:

```
pip install git+https://github.com/ericsomdahl/python-bittrex@master
```

**Comment by eisterman** [comment]
If tomorrow I will make a Python 3 version (and it's sure I will start the work) and send PR this will break the Python 2.7 compatibility.
How should I behave? Creating a new branch for Python 3?

**Comment by ericsomdahl** [comment]
@eisterman, if you wish to do that it is appreciated!

I think maybe the ideal thing would be to use something like Six to support both python versions in the same file.

If you do not wish to mess with that then simply create a bittrex3 module and put the new code there

**Comment by eisterman** [comment]
Some test have resulted in a full compatibility with Python 3 without the need of more efforts.

Can i if the module will be updated on the PyPI with the "Python 3 compatible" flag?
I've done the built in tests with Python 2.7 with and without PyCrypto, and with Python 3.6 with and without PyCrypto and all work flawless

**Comment by mbettan** [comment]
Thanks durika very useful for the workaround!

**Comment by skyl** [comment]
I've been using this exclusively with python3 and haven't noticed any problems.

**Comment by ericsomdahl** [comment]
the setup.py has been updated to reflect current compatibilities

**Comment by awitherow** [comment]
@ericsomdahl can you please link the related PR's so that anyone searching through this history can easily find the changes?

**Comment by awitherow** [comment]
Thank you for your work!

**Comment by ericsomdahl** [comment]
this was addressed in #47
